### PR TITLE
Bug in Weather Units for Broadcasted Notification

### PR DESCRIPTION
**Pull request by skpanagiotis**
This PR resolve Issue number #3419 .

I have added the method `convertWeatherObjectToImperial()` which converts the units of the `notificationPayload` to imperial if needed, in order to pass the object in `sendNotification()`.

### CODE REVIEW

**Comment by khassel** [comment]
Thanks! Can you please add an entry to the `CHANGELOG.md` file? (otherwise the tests will fail ...)

**Comment by khassel** [comment]
if you need help to fix the lint errors in the failed test let me know

**Comment by sdetweil** [comment]
Some of the problem is doing npm run install-mm doesn’t install the git commit hooks

**Comment by skpanagiotis** [comment]
I resolve eslint errors.
Is there a possibility that the failed tests are due to my code?

**Comment by khassel** [comment]
yes, you introduced `const WeatherObject = require("./weatherobject");` but in the tests we have in file `weather_object_spec.js` the same variable name `const WeatherObject = require ...` so renaming one of them should solve this

**Comment by skpanagiotis** [comment]
Thanks very much for your time and help, my mistake is that I skipped running tests, because I was working on Windows system.

File: modules/default/weather/weatherutils.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ const WeatherUtils = {
 	 * Convert temp (from degrees C) into imperial or metric unit depending on
 	 * your config
 	 * @param {number} tempInC the temperature in celsius you want to convert
-	 * @param {string} unit can be 'imperial' or 'metric'
+	 * @param {string} unit can be "imperial" or 'metric'
```
Comment (rejas) [inline modules/default/weather/weatherutils.js:47]: these quote changes should be reverted

File: modules/default/weather/weatherutils.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ const WeatherUtils = {
 	/**
 	 * Convert wind speed into another unit.
 	 * @param {number} windInMS the windspeed in meter/sec you want to convert
-	 * @param {string} unit can be 'beaufort', 'kmh', 'knots, 'imperial' (mph)
+	 * @param {string} unit can be 'beaufort', 'kmh', 'knots, "imperial" (mph)
```
Comment (rejas) [inline modules/default/weather/weatherutils.js:57]: these ones too

**Review by rejas — CHANGES_REQUESTED** [review]

**Review by rejas — CHANGES_REQUESTED** [review]

File: modules/default/weather/weatherutils.js [hunk 3]
```diff
@@ -129,6 +129,39 @@ const WeatherUtils = {
...
+
+	/**
+	 * Convert precipitation value into inch
+	 * @param {number} value the precipitation value for convert
+	 * @param {string} valueUnit can be 'mm' or 'cm'
+	 * @returns {number} the converted precipitation value
+	 */
+	convertPrecipitationToInch (value, valueUnit) {
+		if (valueUnit && valueUnit.toLowerCase() === "cm") return value * 0.3937007874;
+		else return value * 0.03937007874;
```
[the remark this answers is not in the document]
Comment (rejas) [reply modules/default/weather/weatherutils.js:142]: sorry for late reply... 

about cm vs mm: is there some hint that "cm" is used somewhere in the response? weather provider use mm in their data almost exclusivly....

[the remark this answers is not in the document]
Comment (rejas) [reply modules/default/weather/weatherutils.js:142]: sorry for late reply and my first wrong response...

i get your point, will add a small improvement myself and approve

**Review by rejas — APPROVED** [review]

**Pull request merged**